node-data/src/bls.rs

```diff
@@ -139,13 +138,18 @@ fn read_from_file(
     let ciphertext =
         fs::read(path).expect("path should be valid consensus keys file");
 
-    // Decrypt
-    let iv = &ciphertext[..16];
-    let enc = &ciphertext[16..];
+    let mut hasher = Sha256::new();
+    hasher.update(pwd.as_bytes());
+    let hashed_pwd = hasher.finalize().to_vec();
 
-    let cipher =
-        Aes256Cbc::new_from_slices(pwd.as_bytes(), iv).expect("valid data");
-    let bytes = cipher.decrypt_vec(enc).expect("pwd should be valid");
+    let bytes = decrypt(&ciphertext[..], &hashed_pwd).unwrap_or_else(|_| {
+        let hashed_pwd = blake3::hash(pwd.as_bytes());
```

Comment:
I don't like the fact that we keep backward compatibility here. We should get rid of it at some point in my opinion.

Reply:
Agree, that's why there is a warn for the user